### ISSUE TITLE
Multi_proposal_target_layer

### ISSUE THREAD

**Issue by Goingqs** [mahyarnajibi/SNIPER#33]
If I change rpn_post_nms_top_n, the metric will get wrong at 

pred_label = pred.asnumpy().reshape(-1, last_dim).argmax(axis=1).astype('int32')

**Comment by bharatsingh430** [comment]
yes, that is right because we hard coded the number of proposals as 300 in the code to do some shared memory optimizations on GPU. At best we can change it to a macro and you can recompile the code, and right now you need to change in inside the CUDA file.
